pallets/balances/src/lib.rs

```diff
@@ -362,13 +364,17 @@ decl_module! {
             if value.is_zero() { return Ok(()) }
 
             let transactor = ensure_signed(origin)?;
-            let _ = <Self as Currency<_>>::withdraw(
+            let negative_imbalance = <Self as Currency<_>>::withdraw(
```

Comment:
Instead of doubling down on imbalances, we can directly modify `free` balance after calling `ensure_can_withdraw`. It will be a minor optimization. See `transfer` function for usage.

Reply:
I keep this as it is because everywhere we are using `<Self as Currency<_>>::withdraw()` for the token withdrawal then I think it is good to keep consistency.